src/Adapter/Grid/Search/Factory/SearchCriteriaWithCategoryParentIdFilterFactory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Grid\Search\Factory;
+
+use Category;
+use Configuration;
+use Context;
+use PrestaShop\PrestaShop\Core\Grid\Search\Factory\DecoratedSearchCriteriaFactory;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteria;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+use Shop;
+use Tools;
+
+/**
+ * Class SearchCriteriaWithCategoryParentIdFilterFactory
+ */
+final class SearchCriteriaWithCategoryParentIdFilterFactory implements DecoratedSearchCriteriaFactory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createFrom(SearchCriteriaInterface $searchCriteria)
+    {
+        $categoryParentId = $this->resolveCategoryParentId();
+
+        $filters = array_merge(
+            ['id_category_parent' => $categoryParentId],
+            $searchCriteria->getFilters()
+        );
+
+        return new SearchCriteria(
```

Comment:
Also, I'm not against a SearchCriteria in Grid component, we if we do, let's refactor all existing grids and remove the other implementation that exists in Core/Search

Reply:
why do you think we need to remove SearchCriteria implementation from `Core/Search`? Here we have basic `SearchCriteriaInterface` implementation that can be used if needed, but in `Core/Search` we have specific implementations that are more related to Filter component, Grid does not need to know about them, wdyt?

Reply:
I don't think we do, I'm saying I'm not opposed to^^ I agree with you Grids doesn't needs to know about search